src/streamlit_extras/__init__.py

```diff
@@ -0,0 +1,45 @@
+import inspect
+from importlib import import_module
+from typing import Any, Callable, Optional, TypeVar, Union, overload
+
+from streamlit import _gather_metrics
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+@overload
+def extra(
+    func: F,
+) -> F:
+    ...
+
+
+@overload
+def extra(
+    func: None = None,
+) -> Callable[[F], F]:
+    ...
+
+
+def extra(
+    func: Optional[F] = None,
+) -> Union[Callable[[F], F], F]:
+
+    if func:
+
+        filename = inspect.stack()[1].filename
+        extra_name = "streamlit_extras." + filename.split("/")[-2]
+        module = import_module(extra_name)
+
+        if hasattr(module, "__funcs__"):
+            module.__funcs__ += [func]  # type: ignore
+        else:
+            module.__funcs__ = [func]  # type: ignore
+
+        name = f"{module}.{func.__name__}"
+        return _gather_metrics(name=name, func=func)
+
+    def wrapper(f: F) -> F:
```

Comment:
I'm not 100% sure,  but I think you actually need to call `extra` here otherwise `extra` might not work if you would use it like: `@extra()` or maybe if you add parameters later on:

```python
def wrapper(f: F) -> F:
    return extra(func=f)

return wrapper
```

Reply:
I tried adding the `extra()` but quickly testing it looked like an infinite loop then 🤔

Reply:
Will keep this in mind whether we implement more definitions/params in the extra() decorator

Reply:
hmm, but it can't go into the wrapper again since `func` should be not `None` maybe to be extra safe you can put it into an else 🤔